ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -12718,6 +12723,10 @@ bool HqlCppTranslator::requiresTemp(BuildCtx & ctx, IHqlExpression * expr, bool
                 return false;
             return true;
         }
+    case no_alias_scope:
+        if (includeChildren)
+            return true;
```

Comment:
@ghalliday Why should it create temp expressions when  includeChildren==true?

Reply:
@shamser Because if includeChildren is set it means does this expression or any of its children create a temporary expression.  a no_alias_scope will create temporaries for each of the aliases that are associated with it, so should always return true.  Otherwise if you are determining if this expression generates a temporary you need to look at arg(0)